src/stan-users-guide/time-series.qmd

```diff
@@ -634,334 +634,116 @@ time of the original iterations.
 
 ## Hidden Markov models {#hmms.section}
 
-A hidden Markov model (HMM) generates a sequence of $T$ output
-variables $y_t$ conditioned on a parallel sequence of latent
-categorical state variables $z_t \in \{1,\ldots, K\}$.  These
-"hidden" state variables are assumed to form a Markov chain so that
-$z_t$ is conditionally independent of other variables given $z_{t-1}$.
-This Markov chain is parameterized by a transition matrix $\theta$
-where $\theta_k$ is a $K$-simplex for $k \in \{ 1, \dotsc, K \}$.  The
-probability of transitioning to state $z_t$ from state $z_{t-1}$ is
+A Hidden Markov model is a probabilistic
+model over $N$ observations $y_{1:N}$ and $N$ hidden states $z_{1:N}$.
+This models is defined by the conditional distributions $p(y_n \mid z_n, \phi)$
+and $p(z_n \mid z_{n-1}, \phi)$.
+Here we make the dependency on additional model parameters $\phi$ explicit.
+($\phi$ may be a vector of parameters.)
+The complete data likelihood is then
 $$
-z_t \sim \textsf{categorical}(\theta_{z[t-1]}).
+p(y, z \mid \phi) = \prod_n p(y_n \mid z_n, \phi) p(z_n \mid z_{n - 1}, \phi)
 $$
-The output $y_t$ at time $t$ is generated conditionally independently
-based on the latent state $z_t$.
-
-This section describes HMMs with a simple categorical model for
-outputs $y_t \in \{ 1, \dotsc, V \}$.  The categorical distribution for
-latent state $k$ is parameterized by a $V$-simplex $\phi_k$.  The
-observed output $y_t$ at time $t$ is generated based on the hidden
-state indicator $z_t$ at time $t$,
+When $z_{1:N}$ is continuous, the user can explicitly encode these distributions
+in Stan and use Markov chain Monte Carlo to integrate $z$ out.
+
+When each state $z$ takes a value over a discrete and finite set,
+say $\{1, 2, ..., K\}$, we can use Stan's suite of HMM functions to marginalize
+out $z_{1:N}$ and compute
 $$
-y_t \sim \textsf{categorical}(\phi_{z[t]}).
+p(y_{1:N} \mid \phi) = \int_{\mathcal Z} p(y, z \mid \phi) \text d z.
+$$
+We start by defining the conditional observation distribution, stored in a
+$K \times N$ matrix $\omega$ with
+$$
+\omega_{kn} = p(y_n \mid z_n = k, \phi).
+$$
+Next, we introduce the $K \times K$ transition matrix, $\Gamma$, with
+$$
+\Gamma_{ij} = p(z_n = j \mid z_{n - 1} = i, \phi).
+$$
```

Comment:
I think you need to mention that this is a right-stochastic matrix, where we now have that data type built in.